### PR TITLE
pallet epm: add `TrimmingStatus` to the mined solution

### PR DESCRIPTION
**Pull request by niklasad1**
For tools such that is using the `Miner` it's useful to know whether a solution was trimmed or not and also how much that was trimmed.

### CODE REVIEW

File: substrate/frame/election-provider-multi-phase/src/unsigned.rs [hunk 1]
```diff
@@ -108,6 +108,27 @@ impl From<FeasibilityError> for MinerError {
 	}
 }
 
+/// Reports the trimming result of mined solution
+#[derive(Debug, Clone)]
+pub struct TrimmingStatus {
```
Comment (niklasad1) [inline substrate/frame/election-provider-multi-phase/src/unsigned.rs:113]: not sure about naming this, could also be `TrimmingResult` or something.

could also just be a boolean but I prefer a struct here

File: substrate/frame/election-provider-multi-phase/src/unsigned.rs [hunk 2]
```diff
@@ -108,6 +108,27 @@ impl From<FeasibilityError> for MinerError {
...
+	weight: usize,
+	length: usize,
+}
+
+impl TrimmingStatus {
+	pub fn is_trimmed(&self) -> bool {
+		self.weight + self.length > 0
```
Comment (lexnv) [inline substrate/frame/election-provider-multi-phase/src/unsigned.rs:120]: nit: small opt to not rely on the addition and avoid potential edge-cases with overflow 
```suggestion
		self.weight > 0 ||  self.length > 0
```

**Review by lexnv — APPROVED** [review]
LGTM! 👍  Small nit: would be nice to have a test that checks a non-trimmed and a trimmed solution to ensure the numbers are properly propagated. Not sure how easy that is to implement

**Review by lexnv — APPROVED** [review]

**Comment by niklasad1** [comment]
bot merge

**Pull request merged**